src/contract/mod.rs

```diff
@@ -126,7 +126,7 @@ impl<T: Transport> Contract<T> {
   }
 
   /// Call constant function
-  pub fn query<R, A, P>(&self, func: &str, params: P, from: A, options: Options) -> QueryResult<R, T::Out> where
+  pub fn query<R, A, P>(&self, func: &str, params: P, from: A, options: Options, block: Option<BlockNumber>) -> QueryResult<R, T::Out> where
```

Comment:
Can you make it `B: Into<Option<BlockNumber>>`? This allows to omit `Some` when calling the function.

Reply:
Thank you.
I learned a lot!